src/controller/auth.controller.js

```
@@ -0,0 +1,345 @@
+    });
+  } catch (error) {
+    return res.status(500).json({ error: 'Internal server error' });
+  }
+}
+
+async function logout(req, res) {
+  try {
+    const { refreshToken } = req.body;
+
+    if (!refreshToken) {
+      return res.status(400).json({ error: 'Refresh token is required' });
+    }
+
+    await prisma.user.updateMany({
+      where: { refresh_token: refreshToken },
+      data: { refresh_token: null },
+    });
+
+    return res.status(200).json({ message: 'Logged out successfully' });
+  } catch (error) {
+    return res.status(500).json({ error: 'Internal server error' });
+  }
+}
+
+async function refreshAccessToken(req, res) {
+  try {
+    const { refreshToken } = req.body;
+
+    if (!refreshToken) {
+      return res.status(400).json({ error: 'Refresh token is required' });
+    }
+
+    const user = await prisma.user.findUnique({
+      where: { refresh_token: refreshToken },
+    });
+
+    if (!user) {
+      return res.status(403).json({ error: 'Invalid refresh token' });
+    }
+
+    jwt.verify(refreshToken, REFRESH_TOKEN_SECRET, (err, payload) => {
+      if (err) {
+        return res
+          .status(401)
+          .json({ error: 'Invalid or expired refresh token' });
+      }
+
+      const accessToken = generateAccessToken({ userId: user.id });
+
+      return res.status(200).json({ accessToken });
+    });
+  } catch (error) {
+    return res.status(500).json({ error: 'Internal server error' });
+  }
+}
+
+function validatePassword(password) {
+  if (typeof password !== 'string') {
+    return new Error('Password should be a string');
+  }
+
+  if (password.length < 8) {
+    return new Error('Password must be at least 8 characters long');
+  }
+
+  return null;
+}
+
+function validateEmail(email) {
+  if (typeof email !== 'string') {
+    throw new Error('Email should be a string');
+  }
+
+  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+
+  if (!emailRegex.test(email)) {
+    throw new Error('Invalid email format');
```

Inconsistent Error Handling: The `validateEmail` function throws an error instead of returning an error object. For consistency with other validation functions, consider returning an error object.